gamemodes/terrortown/entities/entities/ttt_base_placeable.lua

```diff
@@ -41,6 +43,38 @@ function ENT:SetupDataTables()
     self:NetworkVar("Entity", 0, "Originator")
 end
 
+---
+-- Run if a valid player tries to pick up this entity to check if this pickup is accepted.
+-- @param Player activator The player that used their use key
+-- @return[default=true] boolean Return true to allow pickup
+-- @hook
+-- @realm shared
+function ENT:PlayerCanPickupWeapon(activator)
+    return true
+end
+
+if CLIENT then
+    ---
+    -- Hook that is called if a player uses their use key while focusing on the entity.
+    -- Implement this to predict early if entity can be picked up
+    -- @return bool True to prevent pickup
+    -- @realm client
+    function ENT:ClientUse()
+        local client = LocalPlayer()
+        if not IsValid(client) or not client:IsTerror() or not self.pickupWeaponClass then
+            return true
+        end
+
+        if not self:PlayerCanPickupWeapon(client) then
+            LANG.Msg(client, "pickup_fail", nil, MSG_MSTACK_WARN)
+
+            self:EmitSound(soundDeny)
```

Comment:
there is one thing I'm not so happy here: Previously everyone was able to hear the pickup and deny sound. Now everyone is able to hear the pickup sound, but the deny sound is only audible to the player hittin use. If you could broadcast this easily, it would be really neat.

Reply:
While it sounds lazy, I actually think those sounds shouldnt be broadcasted at all.

Before this rework use was always serverside and therefore those sounds were broadcasted, but I think those are clientside only things. I'd rather remove the sounds completely from being broadcasted.

But what do you think @Histalek @saibotk?

Reply:
tbh i consider this behaviour change a bugfix.

IMO the deny sound is a audible aid for the player trying to pick something up, giving them feedback that the attempt was registered but they are denied. Nothing more.

Making this be heard globally is somewhat unexpected and also makes this part of the deduction process which i think is a mistake.

Reply:
I see your point. While working on the sound stuff yesterday I added a client side sound to body search. This made me wonder if it should be audible to everyone. Hence the question here.

So to be in line with your reasoning, the body search sound should be broadcasted as well I think

Reply:
But for example picking up a weapon is a feedback sound as well. But this is handled by the engine and the sound is audible to everyone.

I just want to get consistency here

Reply:
i don't think there is any consistency to be gotten here. There are many ways to reason about different sounds being globally heard or not.
e.g. weapon pickup is not a feedback sound at all (see CS)

my main point is that something like this shouldn't give any information away about your role/team etc. and in this case forcefully so. (there is no counterplay: if it is your equipment you can't generate the deny sound and you're cooked)

Aside: having bodysearch sounds be audible for all doesn't sound bad.

Reply:
I was mostly asking for us to define a consistent behavior with the limitations we are facing. Thanks

Reply:
FWIW i don't think I've ever heard another player's Use fail sound if it wasn't a scripted map entity button specifically designed to make a "deny" noise